lib/filestreamer.js

```diff
@@ -173,6 +174,14 @@ FileStreamer.prototype._putFile = function (start, callback) {
         callback(err);
     });
     
+    var uploadedSize = 0;
+    file.on('data', function(chunk) {
```

Comment:
This won't report the correct percentage for 2 reasons.

1. The uploaded size is the chunk length.
If I have a 10mb video, each 1mb should report 10%. The first 1mb chunk should report 10%, the second should report 20%. With this logic, if a connection drops after the first mb, and resumes at the start of the second mb, this logic does not distinguish between the second mb and the first mb.

To fix this, you should add the "start" variable to uploadedSize on line 180.

2. The file.on('data') event is part of reading the file, not transmitting the file.

If a connection is dropped, it will ask Vimeo how many bytes were actually received, and resume from there (that's the point of the start variable, it's the new start of the upload). This means an uploader may think it's uploaded more than Vimeo has actually received. 

I have mixed thoughts on fixing this issue. 

1. Changing from file.on('data') to an event fired after the server received the file (some event on the req object, or the pipe below) would be slightly more accurate, because it means there were no network connectivity issues in transmitting the file. This doesn't necessarily mean Vimeo received the file.

2. The most accurate percentage is by asking Vimeo how much of the video ended up on the server with the code below: https://github.com/vimeo/vimeo.js/pull/34/files#diff-3e8ef798ab12c93952db2a3485450bf4R214 but we don't want to spam this with status requests, it will just slow down the upload.

It might be best to offer both of the above as upload percentage options in the long run, but I think option 1. would be a great minor change that would drastically improve the accuracy of this addition!